test/Microsoft.AspNet.Mvc.FunctionalTests/HtmlGenerationTest.cs

```diff
@@ -539,5 +539,39 @@ public class HtmlGenerationTest
             Assert.Equal(expectedContent.Trim(), responseContent, ignoreLineEndingDifferences: true);
 #endif
         }
+
+        [Fact]
+        public async Task EditorTemplateWithNoModel_RendersWithCorrectMetadata()
+        {
+            // Arrange
+            var server = TestHelper.CreateServer(_app, SiteName, _configureServices);
+            var client = server.CreateClient();
+            var expected = "<label class=\"control-label col-md-2\" for=\"Name\">ItemName</label>\r\n" +
+                "<input id=\"Name\" name=\"Name\" type=\"text\" value=\"\" />\r\n\r\n" +
+                "<label class=\"control-label col-md-2\" for=\"Id\">ItemNo</label>\r\n" +
+                "<input data-val=\"true\" data-val-required=\"The ItemNo field is required.\" id=\"Id\" name=\"Id\" type=\"text\" value=\"\" />\r\n\r\n";
+
+            // Act
+            var response = await client.GetStringAsync("http://localhost/HtmlGeneration_Home/ItemUsingSharedEditorTemplate");
+
+            // Assert
+            Assert.Equal(expected, response);
```

Comment:
Need `ignoreLineEndingDifferences: true` here and in next test. This is the reason Travis is currently failing.